pkg/apis/migration/migration.go

```diff
@@ -407,3 +417,61 @@ func (volumeMigration *volumeMigration) registerVolume(ctx context.Context, volu
 	}
 	return volumeID.Id, nil
 }
+
+// cleanupStaleCRDInstances helps in cleaning up stale volume migration CRD instances
+func (volumeMigration *volumeMigration) cleanupStaleCRDInstances() {
+	ticker := time.NewTicker(time.Duration(volumeMigrationInstance.cnsConfig.Global.VolumeMigrationCRCleanupIntervalInMin) * time.Minute)
+	for range ticker.C {
+		ctx, log := logger.GetNewContextWithLogger()
+		config, err := k8s.GetKubeConfig(ctx)
+		if err != nil {
+			log.Warnf("failed to get kubeconfig. err: %v", err)
+			continue
+		}
+		crdGroupClient, err := k8s.NewClientForGroup(ctx, config, CRDGroupName)
+		if err != nil {
+			log.Warnf("failed to create new client for group %q . Err: %+v", CRDGroupName, err)
+			continue
+		}
+
+		log.Infof("Triggering CnsVSphereVolumeMigration cleanup routine")
+		volumeMigrationResourceList := &migrationv1alpha1.CnsVSphereVolumeMigrationList{}
+		err = crdGroupClient.List(ctx, volumeMigrationResourceList)
```

Comment:
One question is instead of creating a new k8sclient, clientForGroup and Listing from API server, can we pass the informer created here - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/apis/migration/migration.go#L165
That informer contains a `Lister()` which i think can be reused..

Reply:
the informer in (https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/apis/migration/migration.go#L165) 
is created inside a go routine(). Let's not reuse that.
Also, I think it's better to recreate k8sclient in cleanupCRDInstances go routine() so that we don't keep the client connection alive through out